pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -300,16 +300,24 @@ func podSumRessourceRequirements(pod *corev1.Pod) corev1.ResourceRequirements {
 				// we need to add it's resources to the main container's resources
 				continue
 			}
-			if limit, ok := c.Resources.Limits[k]; ok {
-				existing := ressourceRequirement.Limits[k]
-				if limit.Cmp(existing) == 1 {
-					ressourceRequirement.Limits[k] = limit
+
+			request, requestFound := c.Resources.Requests[k]
+			limit, limitFound := c.Resources.Limits[k]
+
+			if requestFound {
+				maxRequest := ressourceRequirement.Requests[k]
+				if request.Cmp(maxRequest) == 1 {
+					ressourceRequirement.Requests[k] = request
```

Comment:
The comment is not on this block of code specifiquely, but rather than putting all these branches during computation, would it not be easier to compute requests and limits independently, and then at the end do:
```
if  limitSum != 0 && reqSum > limitSum {
  reqSum = limitSum
}
```

This way we're sure that we never have requests > limits, and don't have to look at every individual containers.
Because to be honest, I struggle already enough to read this code as it is, and I wrote it....

Reply:
Updated with a similar suggestion that also covers init containers. I've also tried to improve readability a bit.
WDYT?

Reply:
Perfect 👌